src/main/java/com/team701/buddymatcher/controllers/users/UserController.java

```diff
@@ -46,4 +43,17 @@ public UserController(UserService userService, ModelMapper modelMapper) {
         }
     }
 
+    @ApiOperation(value = "Put method to update a user's pairingEnabled field")
+    @PutMapping(path = "/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<UserDTO> updatePairingEnabled(@PathVariable("id") Long id, @RequestParam Boolean pairingEnabled) {
```

Comment:
After today's meeting and how we decided the authenticated endpoints will work, the `id` path parameter won't be required right? Since you'll be able to retrieve the `id` of the user through the  [middleware](https://github.com/SE701-T1/backend/issues/57)

What are your thoughts about this?

Reply:
Yeah that's the plan. 

I was wanting to just merge this now, and then add a seperate issue to refactor both this endpoint and the retrieveUserById endpoint once we implement authorisation.

I can also block this PR until we get authorisation merged. 

What do you think is best?

Reply:
Yeah good point actually! We can merge it for now because the endpoint is technically fully functional, and we can change its implementation later once we complete the Auth middleware :)